addons/pos_loyalty/static/src/app/pos_store.js

```diff
@@ -52,7 +52,10 @@ patch(PosStore.prototype, "pos_loyalty.PosStore", {
                 }
             }
         }
-
+        const total_amount = this.globalState.get_order().get_total_with_tax();
+        if (this.globalState.gift_card_product_ids.includes(product.id) && total_amount < 0) {
+            options.price_extra = -product.lst_price - total_amount;
+        }
```

Comment:
I think the business logic to modify is in `_setupGiftCardOptions`. I think that if the product is a `gift_card`, then the logic is supposed to go to that method.

Reply:
The _setupGiftCardOptions methods is there to add a gift card from an existing one, ie when scanning a irl gift card.

Reply:
I debugged and it goes through the method. I think the `addProductFromUi` is generic enough that it takes into account both scanning and manual addition of the gift card product.

Reply:
BTW, when I try to add "eWallet" product, its price doesn't automatically adjust. However, the specs claim it should be:

```
Currently, it's possible to refund an order to a gift card but user has to change the amount manually.
Though, when refunding to an e-wallet, the amount is set to the full refund amount, by default.
```

The point is, if it already does, then you can just reuse the logic that's being used by eWallet, right?